RecoBTag/ONNXRuntime/plugins/DeepCombinedONNXJetTagsProducer.cc

```diff
@@ -0,0 +1,345 @@
+          ++i_output;
+        } else
+          (*(output_tags[flav_n]))[jet_ref] = -2;
+      }
+    }
+  } else {
+    // create empty output collection
+    for (std::size_t i = 0; i < flav_names_.size(); i++) {
+      output_tags.emplace_back(std::make_unique<JetTagCollection>());
+    }
+  }
+
+  // put into the event
+  for (std::size_t flav_n = 0; flav_n < flav_names_.size(); ++flav_n) {
+    iEvent.put(std::move(output_tags[flav_n]), flav_names_[flav_n]);
+  }
+}
+
+void DeepCombinedONNXJetTagsProducer::make_inputs(unsigned i_jet, const reco::DeepFlavourTagInfo& taginfo) {
+  const auto& features = taginfo.features();
+  float* ptr = nullptr;
+  const float* start = nullptr;
+  unsigned offset = 0;
+
+  // jet and other global features
+  offset = i_jet * input_sizes_[kGlobal];
+  ptr = &data_[kGlobal][offset];
+  // jet variables
+  const auto& jet_features = features.jet_features;
+  start = ptr;
+  *ptr = jet_features.pt;
+  *(++ptr) = jet_features.eta;
+  // number of elements in different collections
+  *(++ptr) = features.c_pf_features.size();
+  *(++ptr) = features.n_pf_features.size();
+  *(++ptr) = features.sv_features.size();
+  *(++ptr) = features.npv;
+  // variables from ShallowTagInfo
+  const auto& tag_info_features = features.tag_info_features;
+  *(++ptr) = tag_info_features.trackSumJetEtRatio;
+  *(++ptr) = tag_info_features.trackSumJetDeltaR;
+  *(++ptr) = tag_info_features.vertexCategory;
+  *(++ptr) = tag_info_features.trackSip2dValAboveCharm;
+  *(++ptr) = tag_info_features.trackSip2dSigAboveCharm;
+  *(++ptr) = tag_info_features.trackSip3dValAboveCharm;
+  *(++ptr) = tag_info_features.trackSip3dSigAboveCharm;
+  *(++ptr) = tag_info_features.jetNSelectedTracks;
+  *(++ptr) = tag_info_features.jetNTracksEtaRel;
+  assert(start + n_features_global_ - 1 == ptr);
+
+  // c_pf candidates
+  auto max_c_pf_n = std::min(features.c_pf_features.size(), (std::size_t)n_cpf_);
+  offset = i_jet * input_sizes_[kChargedCandidates];
+  for (std::size_t c_pf_n = 0; c_pf_n < max_c_pf_n; c_pf_n++) {
+    const auto& c_pf_features = features.c_pf_features[c_pf_n];
+    ptr = &data_[kChargedCandidates][offset + c_pf_n * n_features_cpf_];
+    start = ptr;
+    *ptr = c_pf_features.btagPf_trackEtaRel;
+    *(++ptr) = c_pf_features.btagPf_trackPtRel;
+    *(++ptr) = c_pf_features.btagPf_trackPPar;
+    *(++ptr) = c_pf_features.btagPf_trackDeltaR;
+    *(++ptr) = c_pf_features.btagPf_trackPParRatio;
+    *(++ptr) = c_pf_features.btagPf_trackSip2dVal;
+    *(++ptr) = c_pf_features.btagPf_trackSip2dSig;
+    *(++ptr) = c_pf_features.btagPf_trackSip3dVal;
+    *(++ptr) = c_pf_features.btagPf_trackSip3dSig;
+    *(++ptr) = c_pf_features.btagPf_trackJetDistVal;
+    *(++ptr) = c_pf_features.ptrel;
+    *(++ptr) = c_pf_features.drminsv;
+    *(++ptr) = c_pf_features.vtx_ass;
+    *(++ptr) = c_pf_features.puppiw;
+    *(++ptr) = c_pf_features.chi2;
+    *(++ptr) = c_pf_features.quality;
```

Comment:
this could also be a separate function in tensor_filler.cc, later to be integrated in DeepFlavour